packages/core/src/wallet/IndyWallet.ts

```diff
@@ -127,6 +107,8 @@ export class IndyWallet implements Wallet {
         throw new WalletError(errorMessage, { cause: error })
```

Comment:
I know it's not part of your changes but I think we should close the wallet when `createMasterSecret` throws an error, otherwise, we can get into an inconsistent state.